src/Compilers/CSharp/Portable/Symbols/Source/ModifierUtils.cs

```diff
@@ -154,13 +156,27 @@ private static DeclarationModifiers ToDeclarationModifier(SyntaxKind kind)
             }
         }
 
-        public static DeclarationModifiers ToDeclarationModifiers(this SyntaxTokenList modifiers)
+        public static DeclarationModifiers ToDeclarationModifiers(
+            this SyntaxTokenList modifiers, bool allowPartial, DiagnosticBag diagnostics)
         {
             var result = DeclarationModifiers.None;
-
+            bool seenNoDuplicates = true;
```

Comment:
I'm preserving the logic as it was in ParseModifiers.  I think that makes the review a bit easier.